src/controller/CHIPDeviceController.cpp

```diff
@@ -879,6 +879,19 @@ CHIP_ERROR DeviceCommissioner::PairDevice(NodeId remoteDeviceId, RendezvousParam
     VerifyOrExit(mDeviceBeingPaired < kNumMaxActiveDevices, err = CHIP_ERROR_NO_MEMORY);
     device = &mActiveDevices[mDeviceBeingPaired];
 
+    // If the CSRNonce is passed in, using that else using a random one..
+    if (params.HasCSRNonce())
+    {
+        device->SetCSRNonce(params.GetCSRNonce());
```

Comment:
SetCSRNonce returns a CHIP_ERROR. Please use it (likely VerifyOrExit on it)

Reply:
done